src/core.js

```diff
@@ -113,16 +114,29 @@ let Game = {
 
     if (card.get('team') === turn.get('team')) {
       game = game.updateIn(['turn', 'guesses'], guesses => guesses + 1);
+      // Checking if greater than, as opposed to equal to, due to codename's 1 extra guess
+      // if all are guessed correctly.
       if (game.getIn(['turn', 'guesses']) > game.getIn(['turn', 'prompt', 'limit'])) {
         game = game.update('turn', Game.nextTurn);
       }
     } else if (card.get('team') === Game.ASSASSIN) {
-      game = setWinner(game, otherTeam(turn.get('team')));
+      game = Game.setWinner(game, otherTeam(turn.get('team')));
     } else {
       game = game.update('turn', Game.nextTurn);
     }
 
-    // TODO check if someone won
+    // Check if someone won
+    [Game.TEAM_RED, Game.TEAM_BLUE].forEach(team => {
+      let team_has_won = true;
+      game.get('cards').forEach(card => {
+        if (card.get('team') == team && card.get('revealed') == false) {
+          team_has_won = false;
+        }
+      });
+      if (team_has_won == true) {
```

Comment:
Can use [`Arrays.some`](http://stackoverflow.com/questions/2641347/how-to-short-circuit-array-foreach-like-calling-break#2641374) to short-circuit loop.